src/emu/machine.h

```diff
@@ -215,6 +215,7 @@ class running_machine
 
 	// immediate operations
 	int run(bool quiet);
+	int run_frame(attotime frametime=attotime::zero);
```

Comment:
Should this really be here, not protected or something?  Also, default argument values are a source of problems.  The name is also confusing, as it runs one or more scheduler timeslices, not really a "frame" by any definition.